### ISSUE TITLE
How does "saving" actually work?

### ISSUE THREAD

**Issue by mercmobily** [SitePen/dgrid#426]
Hi,

I am going to use _OnDemandList extensively in my application. In fact, it's going to be _the_ widget I will use every single time I have to display data from a store.
Now... I just noticed that _OnDemandList ha,s through _StoreMixin, the following functions:
- updateDirty(id, field, value)
- save()
- revert()

As well as the property getBeforePut.

Now... I don't suppose you explain, in point-list format, how this all works? Raodom questions:
- What is "dirty" data exactly? Do you mean data that has changed locally, but hasn't changed in the remote store?
- What does getBeforePut() imply?
- What happens if there is some "dirty" data (if my understanding is correct), and there is a change in the store while it's observable?
- How would you actually change a row? I know that List()'s row(target) can be used to fetch one. But, how would you _set_ its value, hence (I think) making the item "dirty"?

Basically, I think a short tutorial on how to write a simple JsonRest-backed widget that will display each row, and will allow users to edit a row and save it, explaining in the meantime how to deal with the "observability" and dirty data, would be awesome.

If you give me a few pointers here, even a point-list of things to do/remember and approaches, I will be happy to write it myself and donate it to you guys.

Thank you,

Merc.
